hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -50,7 +50,8 @@ public final class CallerContext {
   public static final String CLIENT_ID_STR = "clientId";
   public static final String CLIENT_CALL_ID_STR = "clientCallId";
   public static final String REAL_USER_STR = "realUser";
-
+  public static final String IS_DFSROUTER = "isDfsRouter";
+  public static final String DFSROUTER_PORT_STR = "dfsRouterPort";
```

Comment:
There is no point having dfsRouterPort or is isDfsRouter, the code is very generic and adding Router specific code should be the last thing to do in the Namenode, Namenode doesn't care  whether router is proxying or any xyz client or service is proxying.

Change it to be generic Proxy Client port nothing exactly router specific in the Namenode